spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -372,6 +405,27 @@ def main(argv: Sequence[str]):
                 im.save(l_fname_out[i])
             display_viewer_syntax(l_fname_out, verbose=verbose)
 
+    # Generate QC report (for `sct_image -stitch` only)
+    if arguments.qc is not None:
+        if arguments.stitch is not None:
+            # NB: The variables `path_tmp`, `fnames_in`, and `im_out_rpi` are generated earlier by `-stitch`
+            printv("Generating QC Report...", verbose=verbose)
+            # specify filenames to use in QC report
+            fname_qc_concat = os.path.join(path_tmp, "concatenated_input_images.nii.gz")
+            fname_qc_out = os.path.join(path_tmp, os.path.basename(fname_out))
+            # generate 2 images to compare in QC report
+            # (1. naively concatenated input images, and 2. stitched image, padded to the larger size of the concat im)
+            ims_in = [Image(fname) for fname in fnames_in]  # NB: `fnames_in` preserves the user's original sorting
+            im_concat, im_out_padded = generate_stitched_qc_images(ims_in, im_out_rpi)
+            im_concat.save(fname_qc_concat)
+            im_out_padded.save(fname_qc_out)
```

Comment:
The commend and variable names are a bit out of date, since we now know that the stitched image is not always smaller than the concatenated image.